ts/Stock/StockToolsBindings.ts

```diff
@@ -499,48 +500,63 @@ extend(NavigationBindings.prototype, {
      * @param {number} defaultHeight
      *        Default height in percents.
      *
+     * @param {boolean} deleteIndicatorAxis
+     *        true, if the indicator is deleted
+     *
      * @return {Array}
      *         An array of calculated positions in percentages.
      *         Format: `{top: Number, height: Number}`
      */
     getYAxisPositions: function (
         yAxes: Array<AxisType>,
         plotHeight: number,
-        defaultHeight: number
+        defaultHeight: number,
+        deleteIndicatorAxis?: boolean
     ): Array<Record<string, number>> {
-        var positions: (Array<Record<string, number>>|undefined),
-            allAxesHeight = 0;
-
+        var positions,
+            allAxesHeight = 0,
+            previousAxisHeight: number;
         /** @private */
-        function isPercentage(prop: (number|string|undefined)): boolean {
+        function isPercentage(prop: number | string | undefined): boolean {
             return defined(prop) && !isNumber(prop) && (prop.match('%') as any);
         }
 
-        positions = yAxes.map(function (
-            yAxis: AxisType
-        ): Record<string, number> {
+        positions = yAxes.map(function (yAxis: AxisType, index: number): Record<string, number> {
             var height = isPercentage(yAxis.options.height) ?
                     parseFloat(yAxis.options.height as any) / 100 :
                     yAxis.height / plotHeight,
                 top = isPercentage(yAxis.options.top) ?
                     parseFloat(yAxis.options.top as any) / 100 :
-                    correctFloat(
-                        yAxis.top - yAxis.chart.plotTop
-                    ) / plotHeight;
+                    correctFloat(yAxis.top - yAxis.chart.plotTop) / plotHeight;
+
+            // New axis' height is NaN so we can check if
+            // the axis is newly created this way
+            if (!deleteIndicatorAxis) {
+                if (!isNumber(height)) {
+                // check if the previous axis is the
+                // indicator axis (every indicator inherits from sma)
+
+                    height = (yAxes as any)[index - 1].series.every((s: Series): boolean => s.is('sma')) ?
```

Comment:
I guess here we might try `yAxes: Array<AxisType>`

Reply:
Probably 'as any' is not necessary at all, Since the yAxes variable has already the ' Array<AxisType>' assigned. Thanks!